lib/paperclip/response_with_limit_adapter.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Paperclip
+  class ResponseWithLimitAdapter < AbstractAdapter
+    def self.register
+      Paperclip.io_adapters.register self do |target|
+        target.is_a?(ResponseWithLimit)
+      end
+    end
+
+    def initialize(target, options = {})
+      super
+      cache_current_values
+    end
+
+    private
+
+    def cache_current_values
+      @original_filename = filename_from_content_disposition || filename_from_path || 'data'
```

Comment:
Is there actually any reason to copy the remote server's filename? What happens if the remote filename does not include a recognized extension?

Reply:
It's not important for any of our current attachments but it might be important in some future use case.

`Attachmentable` concern sets a file extension based on the content type

Reply:
okay… idk, it doesn't seem useful, and could cause issues (mishandling of some encodings for instance).
Also, I guess the upstream server's content-type would be set based on the file extension, so I'm not completely sure what we gain from not just using the file extension?